linux-kernel: update to 6.12.10

KexyBiscuit opened the pull request:
Topic Description
-----------------

- kernel\-tools: update to 6.12.10
    \- Synchronize patches from linux\-kernel.
    Signed\-off\-by: Kexy Biscuit <kexybiscuit@aosc.io>
- linux\-kernel: update to 6.12.10
    \- Routine patch due diligence for upstream releases.
    \- Now following contribution guidelines of Android common kernels <https://android.googlesource.com/kernel/common/+/refs/heads/android\-mainline/README.md>.
    \- Now use B4 shazam to apply patches from list <https://b4.docs.kernel.org/en/latest/maintainer/am\-shazam.html>.
    \- Refresh patch series that have been updated upstream.
    \- Track Loongson patches from Huacai Chen at https://github.com/AOSC\-Tracking/chenhuacai\-linux @ aosc/v6.12, current HEAD is 4260b8dfada390d0f5d4e827947151ea0afcada1.
    \- Track Phytium patches at https://github.com/AOSC\-Tracking/phytium\-linux\-kernel @ aosc/v6.6, current HEAD is 3ea48e9db5214367ad77e786ea91ca6d68227af6.
    \- Track Ubuntu patches at https://github.com/AOSC\-Tracking/ubuntu\-kernel\-next\-linux @ aosc/v6.11, current HEAD is 3c088ba346d5dc0a2861039fee7834626f9eec88.
    \- Track deepin patches at https://github.com/AOSC\-Tracking/deepin\-community\-kernel @ aosc/v6.6, current HEAD is a64f97677a45b8ac6810cbac282fb6393e1b2f0a.
    \- Track deepin rolling patches at https://github.com/AOSC\-Tracking/deepin\-community\-kernel\-rolling @ aosc/v6.12, current HEAD is d85d785db908f29782cb346c09db34b3d58aa024.
    \- Track linux\-surface patches at https://github.com/AOSC\-Tracking/linux\-surface\-kernel @ aosc/v6.12, current HEAD is aee628d0f5e6346e83fe00f5614bb60a7c62cee9.
    \- Track loongarch\-xe patches at https://github.com/AOSC\-Tracking/FanFansfan\-loongson\-linux @ aosc/v6.8, current HEAD is 565064d2079323994aa041f449649918513c7b03.
    \- Track AOSC OS patches at https://github.com/AOSC\-Tracking/linux @ aosc/v6.12.10, current HEAD is d4aa6cec5a4200b06fd7937831c8111e432aefcb.
    \- Add patch for Loongson RTC driver to fix an error when trying to register a clk provider.
    \- Add patch to fix full\-/low\-speed USB devices on ASUS XC\-LS3A6M boards.
    \- Disable loongson driver probing by default for LS7A2000/2K2000 devices.
    Signed\-off\-by: Kexy Biscuit <kexybiscuit@aosc.io>
- kernel\-tools: update to 6.12.10\-rc1
    \- Synchronize patches from linux\-kernel.
    Signed\-off\-by: Kexy Biscuit <kexybiscuit@aosc.io>
- linux+kernel: update to 6.12.10
    Signed\-off\-by: Kexy Biscuit <kexybiscuit@aosc.io>
- linux\-kernel: update to 6.12.10\-rc1
    \- Track AOSC OS patches at https://github.com/AOSC\-Tracking/linux @ aosc/v6.12.10\-rc1, current HEAD is de00965390e5ae497bb104b63da05108ad26c4eb.
    Signed\-off\-by: Kexy Biscuit <kexybiscuit@aosc.io>

Package(s) Affected
-------------------

- kernel-tools: 6.12.10-1
- linux+kernel: 3:6.12.10
- linux-kernel-${__VER}: 1:6.12.10-1

Security Update?
----------------

No

Build Order
-----------

```
#buildit linux-kernel linux+kernel kernel-tools
```

Test Build(s) Done
------------------

**Primary Architectures**

- [x] AMD64 `amd64`
- [x] AArch64 `arm64`
- [x] LoongArch 64-bit `loongarch64`

**Secondary Architectures**

- [x] Loongson 3 `loongson3`
- [x] PowerPC 64-bit (Little Endian) `ppc64el`
- [x] RISC-V 64-bit `riscv64`

KexyBiscuit committed:
linux-kernel: update to 6.12.10-rc1

- Track AOSC OS patches at https://github.com/AOSC-Tracking/linux @ aosc/v6.12.10-rc1, current HEAD is de00965390e5ae497bb104b63da05108ad26c4eb.

Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>

KexyBiscuit committed:
linux+kernel: update to 6.12.10

Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>

KexyBiscuit committed:
kernel-tools: update to 6.12.10-rc1

- Synchronize patches from linux-kernel.

Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>

MingcongBai reviewed (dismissed):
LGTM (no upstream regression report, yet).

KexyBiscuit committed:
linux-kernel: update to 6.12.10

- Routine patch due diligence for upstream releases.
  - Now following contribution guidelines of Android common kernels <https://android.googlesource.com/kernel/common/+/refs/heads/android-mainline/README.md>.
  - Now use B4 shazam to apply patches from list <https://b4.docs.kernel.org/en/latest/maintainer/am-shazam.html>.
  - Refresh patch series that have been updated upstream.
  - Track Loongson patches from Huacai Chen at https://github.com/AOSC-Tracking/chenhuacai-linux @ aosc/v6.12, current HEAD is 4260b8dfada390d0f5d4e827947151ea0afcada1.
  - Track Phytium patches at https://github.com/AOSC-Tracking/phytium-linux-kernel @ aosc/v6.6, current HEAD is 3ea48e9db5214367ad77e786ea91ca6d68227af6.
  - Track Ubuntu patches at https://github.com/AOSC-Tracking/ubuntu-kernel-next-linux @ aosc/v6.11, current HEAD is 3c088ba346d5dc0a2861039fee7834626f9eec88.
  - Track deepin patches at https://github.com/AOSC-Tracking/deepin-community-kernel @ aosc/v6.6, current HEAD is a64f97677a45b8ac6810cbac282fb6393e1b2f0a.
  - Track deepin rolling patches at https://github.com/AOSC-Tracking/deepin-community-kernel-rolling @ aosc/v6.12, current HEAD is d85d785db908f29782cb346c09db34b3d58aa024.
  - Track linux-surface patches at https://github.com/AOSC-Tracking/linux-surface-kernel @ aosc/v6.12, current HEAD is aee628d0f5e6346e83fe00f5614bb60a7c62cee9.
  - Track loongarch-xe patches at https://github.com/AOSC-Tracking/FanFansfan-loongson-linux @ aosc/v6.8, current HEAD is 565064d2079323994aa041f449649918513c7b03.
  - Track AOSC OS patches at https://github.com/AOSC-Tracking/linux @ aosc/v6.12.10, current HEAD is d4aa6cec5a4200b06fd7937831c8111e432aefcb.
- Add patch for Loongson RTC driver to fix an error when trying to register a clk provider.
- Add patch to fix full-/low-speed USB devices on ASUS XC-LS3A6M boards.
- Disable loongson driver probing by default for LS7A2000/2K2000 devices.

Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>

KexyBiscuit committed:
kernel-tools: update to 6.12.10

- Synchronize patches from linux-kernel.

Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>

KexyBiscuit commented:
Smoke testing passed on AMD64 generation 2, AMD64 generation 1 and Armv8-A generation 2 Hyper-V VMs.

![image](https://github.com/user-attachments/assets/3c584cb0-926d-4cc0-8d2d-ced987b60c97)
![image](https://github.com/user-attachments/assets/6a427c76-bb7e-47ec-976a-9f660e8f44cc)
![image](https://github.com/user-attachments/assets/6fc78367-7fe2-4159-b2ef-6f3328986679)

KexyBiscuit commented:
Smoke testing passed on AMD64 VMware VM.

- Product: VMware® Workstation 17 Pro
- Version: 17.6.2 build-24409262
- Host OS version: Windows 11 Enterprise, 64-bit (Build 27774.1000) 10.0.27774
- Accelerate 3D graphics: on
- Firmware type: UEFI
- Host VBS Mode: on

![image](https://github.com/user-attachments/assets/3d2d0aa0-552e-4a07-88d0-d2b5966e2e72)

KexyBiscuit commented:
Smoke testing passed on XA612A0 with LG110.

![image](https://github.com/user-attachments/assets/ada95b20-3b3a-4f70-8ad3-05ce8b149011)
